src/main/java/org/hydev/mcpm/client/injector/PluginLoader.java

```diff
@@ -1,45 +1,135 @@
 package org.hydev.mcpm.client.injector;
 
+import com.google.common.reflect.TypeToken;
+import org.bukkit.Bukkit;
+import org.bukkit.command.Command;
+import org.bukkit.command.PluginCommand;
+import org.bukkit.command.SimpleCommandMap;
+import org.bukkit.event.Event;
+import org.bukkit.plugin.InvalidDescriptionException;
+import org.bukkit.plugin.InvalidPluginException;
+import org.bukkit.plugin.Plugin;
+import org.bukkit.plugin.RegisteredListener;
+import org.hydev.mcpm.utils.PluginJarFile;
+
 import java.io.File;
+import java.io.IOException;
+import java.net.URLClassLoader;
+import java.util.*;
+
+import static org.hydev.mcpm.utils.ReflectionUtils.getPrivateField;
+import static org.hydev.mcpm.utils.ReflectionUtils.setPrivateField;
 
 /**
  * TODO: Write a description for this class!
  *
  * @author Azalea (https://github.com/hykilpikonna)
  * @since 2022-09-27
  */
-public class PluginLoader
+public class PluginLoader implements LoadBoundary, UnloadBoundary, ReloadBoundary
 {
-    /**
-     * Dynamically load a local plugin through JVM reflections and classloader hacks
-     *
-     * @param jar Local jar file path
-     */
+    @Override
+    public void loadPlugin(String name) throws PluginNotFoundException
+    {
+        // 1. Find plugin file by name
+        var nf = new PluginNotFoundException(name);
+        var dir = new File("plugins");
+        if (!dir.isDirectory()) throw nf;
+        var file = Arrays.stream(Optional.ofNullable(dir.listFiles()).orElseThrow(() -> nf))
+            .filter(f -> f.getName().endsWith(".jar"))
+            .filter(f -> {
+                try (var jf = new PluginJarFile(f))
+                {
+                    return jf.readPluginYaml().getName().equalsIgnoreCase(name);
+                }
+                catch (IOException ignored) { return false; }
+            }).findFirst().orElseThrow(() -> nf);
+
+        loadPlugin(file);
+    }
+
+    @Override
     public void loadPlugin(File jar)
     {
-        // TODO: Implement this
-        throw new UnsupportedOperationException("TODO");
+        // 2. Load plugin
+        var pm = Bukkit.getPluginManager();
+        try
+        {
+            var plugin = pm.loadPlugin(jar);
+            assert plugin != null;
```

Comment:
As a caller I'd probably want to know if loadPlugin failed (and I'd probably be able to say something nice to our user instead of panicking).

I think it's a good idea to make this return a discardable boolean or return some kind of failure indicator along with the assert (I'm talking about the catch stmt below too), but I understand if this is a rare case and you just want to leave it like this.